Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py

```diff
@@ -59,11 +66,34 @@ def parser(cls, parser, loggers=None):
         )
 
     @classmethod
-    def revert_commit(cls, args, repository, **kwargs):
-        # Check if there are any outstanding changes:
-        if repository.modified():
-            sys.stderr.write('Please commit your changes or stash them before you revert commit: {}'.format(args.commit))
-            return 1
+    def get_issue_info(cls, args, repository, **kwargs):
```

Comment:
This is directly lifted from `program/branch.py`, yes? Have we looked at combining the logic?

Reply:
In regards to consolidating with `branch.py`, there's a problem with using the existing infra with branchpoint/branch/PR creation:

We need to set an issue before calling PullRequest.pull_request_branch_point or else that code will set args.issue from the commit we want to revert and create a PR using that.

Right now the issue logic in `branch.py` is spread over its `main()`. So maybe the best solution is to throw all that in a function and then call it in `revert.py`. (As discussed, will be doing that in a separate change.)